process_pr.py

```diff
@@ -1254,38 +1348,117 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
                         override_tests_failure = reason
                         if not dryRun:
                             set_comment_emoji(comment.id, repository)
+    # end of parsing comments section
 
-        # Check L2 signoff for users in this PR signing categories
-        if [x for x in commenter_categories if x in signing_categories]:
-            ctype = ""
-            selected_cats = []
-            if re.match("^([+]1|approve[d]?|sign|signed)$", first_line, re.I):
-                ctype = "+1"
-                selected_cats = commenter_categories
-            elif re.match("^([-]1|reject|rejected)$", first_line, re.I):
-                ctype = "-1"
-                selected_cats = commenter_categories
-            elif re.match("^[+-][a-z][a-z0-9-]+$", first_line, re.I):
-                category_name = first_line[1:].lower()
-                if category_name in commenter_categories:
-                    ctype = first_line[0] + "1"
-                    selected_cats = [category_name]
-            if ctype == "+1":
-                for sign in selected_cats:
-                    signatures[sign] = "approved"
-                    if (test_comment is None) and (
-                        (repository in auto_test_repo) or ("*" in auto_test_repo)
-                    ):
-                        test_comment = comment
-                    if sign == "orp":
-                        mustClose = False
-            elif ctype == "-1":
-                for sign in selected_cats:
-                    signatures[sign] = "rejected"
-                    if sign == "orp":
-                        mustClose = False
+    if (
+        (not warned_too_many_commits)
+        and not ok_too_many_commits
+        and pr.commits >= MAX_INITIAL_COMMITS_IN_PR
+    ):
+        if not dryRun:
+            issue.create_comment(
+                f"This PR contains too many commits ({pr.commits} > {MAX_INITIAL_COMMITS_IN_PR}). "
+                "Make sure you chose the right target branch.\n"
+                f"{l2s}, you can override this check with `+commit-count`."
+            )
+        return
 
-    # end of parsing comments section
+    # Get the commit cache from `already_seen` commit or technical commit
+    print("Checking for commit cache")
+    cache_comment = None
+
+    if technical_comment:
+        cache_comment = technical_comment
+    else:
+        if already_seen:
+            cache_comment = already_seen
+
+    if cache_comment:
+        seen_commits_match = REGEX_COMMITS_CACHE.search(cache_comment.body)
+        if seen_commits_match:
+            print("Loading commit cache")
+            commit_cache = loads(seen_commits_match[1])
+
+    if pr.commits < MAX_INITIAL_COMMITS_IN_PR or ok_too_many_commits:
+        for commit in all_commits:
+            if commit.sha not in commit_cache:
+                commit_cache[commit.sha] = {
```

Comment:
@iarspider , we should call `api_rate_limits_repo` here to slow down the processing if api rate is reaching to 0

Reply:
Done

Reply:
not really, it should be added in the `for` loop under the `if commit.sha not in commit_cache:` condition